src/content.js

```diff
@@ -148,20 +149,32 @@ export default class Content {
   }
 
   /**
-   * Replace templated variables in the provided content if requested.
-   * @param {Config} config
-   * @param {string} input - The initial value of the content.
-   * @returns {string} Content with templatized variables replaced.
+   * Replace templated variables in the provided content as requested.
+   * @param {unknown} input - The initial value of the content.
+   * @returns {unknown} Content with templatized variables replaced.
    */
-  templatize(config, input) {
-    if (!config.inputs.payloadTemplated) {
-      return input;
+  templatize(input) {
+    if (Array.isArray(input)) {
+      return input.map((v) => this.templatize(v));
+    }
+    if (input && typeof input === "object") {
+      /**
+       * @type {any}
+       */
```

Comment:
Not very familiar with this typing format in JS doc but would we want to be more specific but using a `Record<string, string>`?